src/csmake.pl

```diff
@@ -53,31 +54,43 @@
 }
 $instdir = abs_path($instdir);
 
-my $extract_targets = 0;
+# Copy arguments into MAKEARGS to use them with real make
+my @MAKEARGS = @ARGV;
 
-# Check CSMAKEFLAGS environment variable
+# Expand command-line arguments to include the CSMAKEFLAGS environment variable
 if ($ENV{'CSMAKEFLAGS'}) {
 	@ARGV = (split(/\s+/, $ENV{'CSMAKEFLAGS'}), @ARGV);
 }
 
 # Parse arguments
-if (@ARGV > 0 && $ARGV[0] eq "-t") {
-	$extract_targets = 1;
-	shift @ARGV;
+my %options=();  # csmake options
+{
+    local $SIG{__WARN__} = sub { };  # Supress warnings
+    getopts("hN:T:", \%options);
+    # Remove csmake options from MAKEARGS
+    my $i=0;
```

Comment:
Wouldn't it be better to push them to `@ARGV` and let `getopts` deal with them?

Reply:
Which one to push?

Reply:
`CSMAKEARGS`

Reply:
No, this way we would lose MAKE's arguments.